### PR TITLE
specs, add test case to type/property/value-types, property has type of enum value

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Real case in OpenAI (currently it is value from Enum; it should be changed to value from Union, upon [PR](https://github.com/Azure/azure-rest-api-specs/pull/27687/files#diff-d40c5083c3e6eab3b0a599ba87fb01d60283f82fb428e0f5014a5a3654c16603))
https://github.com/Azure/azure-rest-api-specs/blob/ca6cb3595fb0d091691d09e76594546c01b074ea/specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/azure_cognitive_search_options.tsp#L22

# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/type/property/value-types/main.tsp [hunk 1]
```diff
@@ -269,3 +269,19 @@ model UnionFloatLiteralProperty is ModelTemplate<43.125 | 46.875>;
...
+
+union BaseEnum {
+  string,
+  EnumValue1: "value1",
+}
+
+union ExtendedEnum {
+  BaseEnum,
+  EnumValue2: "value2",
+}
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:281]: I use the extended enum, as dev could write `property: BaseEnum.EnumValue1` as well.

Guess we first get the basic case covered.

File: packages/cadl-ranch-specs/http/type/property/value-types/main.tsp [hunk 2]
```diff
@@ -144,13 +144,13 @@ enum FixedInnerEnum {
 model EnumProperty is ModelTemplate<FixedInnerEnum>;
 @route("/enum")
 @operationGroup
-interface Enum extends ModelOperations<EnumProperty, "ValueOne"> {}
+interface Enum extends ModelOperations<EnumProperty, "\"ValueOne\""> {}
```
Comment (msyyc) [inline packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:147]: Does the change only influence summary.md?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:147]: Yes, this part on `ValueOne` and `UnknownValue` is only for the fix on summary doc.

File: packages/cadl-ranch-specs/http/type/property/value-types/main.tsp (hunk 1 above)
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:281]: Seems weird. Why we don't use:
```
union BaseEnum {
  string,
  EnumValue1: "value1",
  EnumValue2: "value2",
}
```
directly?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:281]: Dev may want to "extend" a Enum.

And it is already in use
https://github.com/Azure/azure-rest-api-specs/pull/27749#discussion_r1494047254

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:281]: Checked the logic for `getUnionAsEnum` and it support such defintion.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:281]: Yeah, that is the "flatten" on `flattenedMembers`

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp]: I've removed the hierarchy in this case, as the expected behavior is in discussion.

**Review by tadelesh — APPROVED** [review]

**Review by MaryGao — APPROVED** [review]
LGTM

**Review by pshao25 — APPROVED** [review]

**Pull request merged**